packages/tailwindcss/src/ast.ts

```diff
@@ -1,9 +1,20 @@
-export type Rule = {
+import { parseAtRule } from './css-parser'
+
+const AT_SIGN = 0x40
+
+export type StyleRule = {
```

Comment:
I think we should just use `Rule` for this type and `rule` for the helper function instead of `styleRule`, as rules and at-rules are distinct concepts in CSS terminology already:

<img width="853" alt="image" src="https://github.com/user-attachments/assets/83d93f9f-dec6-40f0-baee-5a48c9493fec">

Reply:
I was thinking the same, but @thecrypticace pointed me to the spec where they use Style rules and At rules. I think _just_ Rule is very common because of PostCSS.

- https://www.w3.org/TR/css-syntax-3/#css-stylesheets
 
<img width="246" alt="image" src="https://github.com/user-attachments/assets/61777034-1bb1-47eb-af69-256e7375b1a5">

I'm fine with either. Biggest benefit for separating is that we have a `rule()` abstraction that returns a style rule via `styleRule()` or an at rule via `atRule()` right now depending on the `@` being present.

Reply:
Yeah the spec *explicitly* calls them style rules.

Reply:
Yeah I'm fine with style rule then if that's considered more technically correct 👍